### PR TITLE
Update intro about reliable/unreliable data

### LINKED ISSUE #56: Is reliability really stream-based?
In the introduction, it says
"Reliability within QUIC is performed on a per-stream basis, so some frame types are not eligible for retransmission"

Is it really per-stream? Some frames are retransmitted even though they aren't associated with a stream.

Really, the detection mechanism is packet-based, the retransmission decision is made on a per-frame basis, and for stream-specific frames there is the added consideration of stream state (i.e. do not resend if reset)

I suggest replacing the sentence above with "Some QUIC frame types are not eligible for retransmission."

### PR DESCRIPTION
**Pull request by tfpauly**
Closes #56

### CODE REVIEW

**Review by martinduke — APPROVED** [review]
LGTM

File: draft-ietf-quic-datagram.md [hunk 1]
```diff
@@ -57,8 +57,9 @@ Discussion of this work is encouraged to happen on the QUIC IETF mailing list
 
 The QUIC Transport Protocol {{!RFC9000}} provides a secure,
 multiplexed connection for transmitting reliable streams of application data.
-Reliability within QUIC is performed on a per-stream basis, so some frame
-types are not eligible for retransmission.
+QUIC uses various frame types to transmit data within packets, and each frame
+type defines whether or not the data it contains will be retransmitted.
+Streams of reliable application data are sent using STREAM frames.
```
Comment (LPardue) [inline draft-ietf-quic-datagram.md:62]: change look fine but is pretty duplicative of L74-75. Maybe remove that text?

Comment (DavidSchinazi) [reply draft-ietf-quic-datagram.md:62]: I agree with Lucas, the new text makes L74-75 redundant so I'd remove L74-75

**Review by DavidSchinazi — APPROVED** [review]
LGTM modulo one comment

File: draft-ietf-quic-datagram.md (hunk 1 above)
Comment (tfpauly) [reply draft-ietf-quic-datagram.md:62]: Sounds good, removed those redundant lines.

**Pull request merged**